### PR TITLE
feat: add mapping search api endpoint

### PR DESCRIPTION
**Pull request by matthewBearCamunda**
## Description

Creates an endpoint for searching mapping rules

## Checklist

- [ ] for CI changes:
  - [ ] structural/foundational changes signed off by [CI DRI](https://github.com/cmur2)
  - [ ] [ci.yml](https://github.com/camunda/camunda/blob/main/.github/workflows/ci.yml) modifications comply with ["Unified CI" requirements](https://github.com/camunda/camunda/wiki/CI-&-Automation#workflow-inclusion-criteria)

## Related issues

related to #22392

### CODE REVIEW

File: zeebe/gateway-protocol/src/main/proto/rest-api.yaml [hunk 1]
```diff
@@ -2606,6 +2606,42 @@ paths:
         "500":
           $ref: "#/components/responses/InternalServerError"
 
+  /mappings/search:
```
Comment (Ben-Sheppard) [inline zeebe/gateway-protocol/src/main/proto/rest-api.yaml:2609]: ❓ in here we are using `/mappings` but in the controller its `mapping-rules`, we should align these, my vote for now would be `/mappings`

File: zeebe/gateway-protocol/src/main/proto/rest-api.yaml [hunk 2]
```diff
@@ -2606,6 +2606,42 @@ paths:
...
+        - Mapping rule
+      operationId: findMappings
+      summary: Query mappings
+      description: |
+        Search for mapping rules based on given criteria.
+      requestBody:
+        content:
+          application/json:
+            schema:
+              $ref: "#/components/schemas/UserSearchQueryRequest"
```
Comment (Ben-Sheppard) [inline zeebe/gateway-protocol/src/main/proto/rest-api.yaml:2621]:
```suggestion
              $ref: "#/components/schemas/MappingSearchQueryRequest"
```

File: zeebe/gateway-protocol/src/main/proto/rest-api.yaml [hunk 3]
```diff
@@ -3387,6 +3423,33 @@ components:
...
+      description: Mapping rule search response item.
+      type: object
+      properties:
+        mappingKey:
+          type: integer
+          description: The unique system-generated internal mapping ID.
+          format: int64
+        claimName:
+          type: string
+          description: The claim name, idk.
```
Comment (Ben-Sheppard) [inline zeebe/gateway-protocol/src/main/proto/rest-api.yaml:3436]:
```suggestion
          description: The claim name to match against a token.
```

File: zeebe/gateway-protocol/src/main/proto/rest-api.yaml [hunk 4]
```diff
@@ -3387,6 +3423,33 @@ components:
...
+        claimValue:
+          type: string
+          description: The claim value, idk.
```
Comment (Ben-Sheppard) [inline zeebe/gateway-protocol/src/main/proto/rest-api.yaml:3439]:
```suggestion
          description: The value of the claim to match.
```

File: zeebe/gateway-protocol/src/main/proto/rest-api.yaml (hunk 1 above)
Comment (matthewBearCamunda) [reply zeebe/gateway-protocol/src/main/proto/rest-api.yaml]: I'd prefer to keep `mapping-rules` here as its used in the main `MappingController`, as well as in `TenantController`

File: zeebe/gateway-protocol/src/main/proto/rest-api.yaml [hunk 5]
```diff
@@ -2675,6 +2675,42 @@ paths:
         "500":
           $ref: "#/components/responses/InternalServerError"
 
+  /mapping-ruless/search:
```
Comment (Ben-Sheppard) [inline zeebe/gateway-protocol/src/main/proto/rest-api.yaml:2678]:
```suggestion
  /mapping-rules/search:
```

**Review by Ben-Sheppard — COMMENTED** [review]
Just one thing but I'll approve early as the rest seem good to me

File: zeebe/gateway-rest/src/main/java/io/camunda/zeebe/gateway/rest/SearchQueryRequestMapper.java [hunk 6]
```diff
@@ -675,6 +702,22 @@ private static List<String> applyTenantSortField(
...
   }
 
+  private static List<String> applyMappingSortField(
+      final String field, final MappingSort.Builder builder) {
+    final List<String> validationErrors = new ArrayList<>();
+    if (field == null) {
+      validationErrors.add(ERROR_SORT_FIELD_MUST_NOT_BE_NULL);
+    } else {
+      switch (field) {
+        case "mappingKey" -> builder.mappingKey();
```
Comment (Ben-Sheppard) [inline zeebe/gateway-rest/src/main/java/io/camunda/zeebe/gateway/rest/SearchQueryRequestMapper.java:712]: ❓ Sorry I didn't pick this up before, with other indexes we've had an issue in searching in that the search request expects there to be a `key` field, I believe its used as a default sorting etc. Can you confirm that if you run this locally and search the request passes?

**Review by Ben-Sheppard — APPROVED** [review]

**Pull request merged**